framework/include/outputs/PNGObject.h

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Forward declarations
+class PNGObject;
+
+template <>
+InputParameters validParams<PNGObject>();
+
+// MOOSE includes
+#include "Output.h"
+class PNGObject : public Output
+{
+public:
+  //Basic constructor.  Takes parameters passed in to create a PNGCreator object.
+  PNGObject(const InputParameters & parameters);
+
+  // Method for assigning color values to the PNG
+  void setRGB(png_byte *rgb, float selection);
```

Comment:
How is this working? You are not including libpng stuff here but use this type in a public interface. Does it have to be public in the first place?

Reply:
Probably doesn't need to be public. This PR has a ways to go. I'm helping Sam get the right things into the Makefile and then we'll continue to work on the interface.

Reply:
Yeah, I don't see a reason any of it should be public except the constructor.  Public was just easier to start.  Right now I switched everything except the constructor to protected.  Not sure if any of it should be private yet.  I haven't really thought about what other classes would need from this class, if they'd ever need anything at all.